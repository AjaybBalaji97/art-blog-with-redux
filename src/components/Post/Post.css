.allPosts{
    margin-top: 20px;
}
PostReaction{
    bottom: 0;
}