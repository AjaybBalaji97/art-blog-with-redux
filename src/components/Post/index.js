import React, { Component } from 'react';
import PostContent from '../PostContent';
import PostReaction from '../PostReaction';
import './Post.css';
class Post extends Component {
    
      state={
          liked:this.props.product.liked,
          claps:this.props.product.claps,
      }
      handleLikeClick=()=>{
        this.setState({
            liked: this.state.liked?false:true
          });
      }
      handleClapClick=()=>{
        this.setState({
            claps:this.state.claps+1
          });
      }
    render() {
        const product = this.props.product;

      return (
          <div className="allPosts">
            <PostContent product={product}/>
            <PostReaction liked={this.state.liked} claps={this.state.claps} handleLikeClick={this.handleLikeClick} handleClapClick={this.handleClapClick}/>
            
          </div>
      );
    }
  }
  export default Post;