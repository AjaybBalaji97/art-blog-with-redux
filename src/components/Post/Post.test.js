import React from 'react';
import renderer from 'react-test-renderer';
import Post from '.';
import {shallow} from 'enzyme';


describe('PostContent',()=>{
    const products={
        "date": "2nd Januray, 2018",
        "readingTime": "2 mins",
        "title": "The future of abstract art and the culture ...",
        "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
        "claps": 10,
        "liked": false,
        "image": "abstract.png"
    }
    it('should render Post contents',()=>{
        
        const tree=renderer.create(<Post product={products} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should initialize a state variable with value same as the props',()=>{
        const wrapper=shallow(<Post  product={products}/>);
        //wrapper.find('counter').simulate('click');
        expect(wrapper.instance().state.liked).toEqual(false);
        expect(wrapper.instance().state.claps).toEqual(10);
    })
    it('should increment the counter on "incrementCount" function',()=>{
        const wrapper=shallow(<Post  product={products}/>);
        wrapper.instance().handleClapClick()
        expect(wrapper.instance().state.claps).toEqual(11)
    })
    it('should decrement the counter on "decrementCount" function',()=>{
        const wrapper=shallow(<Post  product={products}/>);
        wrapper.instance().handleLikeClick()
        expect(wrapper.instance().state.liked).toEqual(true)
    })

});