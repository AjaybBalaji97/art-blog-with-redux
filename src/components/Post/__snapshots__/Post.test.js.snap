// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContent should render Post contents 1`] = `
<div
  className="allPosts"
>
  <div
    className="main"
  >
    <div
      className="backgroundImage"
      style={
        Object {
          "backgroundImage": "url(\\"./Images/abstract.png\\")",
        }
      }
    />
    <div
      className="imageInfo"
    >
      <span>
        2nd Januray, 2018
      </span>
      <span>
        2 mins
      </span>
    </div>
    <div
      className="imageDesc"
    >
      <span>
        The future of abstract art and the culture ...
      </span>
      <span
        className="imageContent"
      >
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </span>
    </div>
    <br />
  </div>
  <div
    className="main"
  >
    <span>
      <img
        alt="clap"
        className="clap"
        onClick={[Function]}
        src="./Icons/clapping.svg"
      />
      <p>
        10
      </p>
      <img
        alt="heart"
        className="heart"
        onClick={[Function]}
        src="./Icons/heart-black.svg"
      />
    </span>
  </div>
</div>
`;
